neqo-http3/src/server.rs

```diff
@@ -230,21 +283,36 @@ fn prepare_data(
     handler: &HandlerRef,
     now: Instant,
     events: &mut Http3ServerEvents,
+    wt: bool,
 ) {
     loop {
         let mut data = vec![0; MAX_EVENT_DATA_SIZE];
-        let res =
-            handler_borrowed.read_request_data(&mut conn.borrow_mut(), now, stream_id, &mut data);
+        let res = handler_borrowed.read_request_data(
+            &mut conn.borrow_mut(),
+            now,
+            stream_id,
+            &mut data,
+            wt,
+        );
+        qtrace!("prepare_data read stream_id:{} res={:?}", stream_id, res);
         if let Ok((amount, fin)) = res {
             if amount > 0 {
                 if amount < MAX_EVENT_DATA_SIZE {
                     data.resize(amount, 0);
                 }
-                events.data(
-                    ClientRequestStream::new(conn.clone(), handler.clone(), stream_id),
-                    data,
-                    fin,
-                );
+                if !wt {
+                    events.data(
+                        ClientRequestStream::new(conn.clone(), handler.clone(), stream_id),
+                        data,
+                        fin,
+                    );
+                } else {
+                    events.web_transport_stream_data(
```

Comment:
I wonder if events.data() might be made generic so that it could take a webtransport event.  You would need to make this function take generic also, so maybe it's not worth it (`FnMut(&mut ActiveConnectionRef, &HandlerRef, u64) -> impl DataEvent`) but all these "is webtransport" checks are not a good sign as it makes the code much harder to reason about.